core/src/main/java/org/opensearch/remote/metadata/client/impl/LocalClusterIndicesClient.java

```diff
@@ -101,20 +99,48 @@ public CompletionStage<PutDataObjectResponse> putDataObjectAsync(
         Executor executor,
```

Comment:
Do we still need executor here then?

Reply:
It's part of the interface we're implementing, so yes for now